src/charm.py

```diff
@@ -105,6 +106,7 @@ def __init__(self, *args):
         self._certificates = TLSCertificatesRequiresV3(self, "certificates")
         self._logging = LogForwarder(charm=self, relation_name=LOGGING_RELATION_NAME)
         self.unit.set_ports(NRF_SBI_PORT)
+        self.framework.observe(self.on.collect_unit_status, self._on_collect_unit_status)
```

Comment:
If we add this line before checking unit leadership, scaled units status will be set as expected.